stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -3357,12 +3357,13 @@ impl StacksMessageCodec for NakamotoBlock {
     }
 
     fn consensus_deserialize<R: std::io::Read>(fd: &mut R) -> Result<Self, CodecError> {
-        let (header, txs) = {
+        let header: NakamotoBlockHeader = read_next(fd)?;
+
+        let txs: Vec<StacksTransaction> = {
             let mut bound_read = BoundReader::from_reader(fd, u64::from(MAX_MESSAGE_LEN));
-            let header: NakamotoBlockHeader = read_next(&mut bound_read)?;
-            let txs: Vec<_> = read_next(&mut bound_read)?;
-            (header, txs)
-        };
+            // The latest epoch where StacksMicroblock exist is Epoch25
+            read_next_at_most(&mut bound_read, u32::MAX)
```

Comment:
You can use `read_next()` here because `BoundReader` will stop you from reading more than `MAX_MESSAGE_LEN` bytes